core/embed/trezorhal/stm32u5/consumption_mask.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include STM32_HAL_H
+#include "rng.h"
+
+#define SAMPLES 110
+#define TIMER_PERIOD 16640  // cca 10 KHz @ 180MHz
```

Comment:
fixed https://github.com/trezor/trezor-firmware/pull/3728/commits/a5ea7cb20e75f9cedc02d3095f1a9a6b59be21e3

still close enough to 10 kHz.. (note that this will undergo further tuning and i will change the constants if needed)

Also removed the special placement of the pwm_data, not needed on U5 as all ram can be accessed by DMA